contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,473 @@
+    Capacity string `json:"capacity"`
+}
+
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
+	Phase string `json:"phase"`
+}
+
+// CStorPoolClusterSpecAttr is to describe zpool related attributes.
+type CStorPoolClusterSpecAttr struct {
+	CacheFile        string   `json:"cacheFile"`        //optional, faster if specified
+	PoolType         PoolType `json:"poolType"`         //mirrored, striped, raidz or raidz2
+	OverProvisioning bool     `json:"overProvisioning"` //true or false
+}
+
+
+```
+
+#### Spec for CStorPool
+TODO: Put sample YAMLs
+
+```
+
+
+// CStorPool describes a cstor pool resource created as custom resource.
+type CStorPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CStorPoolSpec   `json:"spec"`
+	Status CStorPoolStatus `json:"status"`
+}
+
+// CStorPoolSpec is the spec listing fields for a CStorPool resource.
+type CStorPoolSpec struct {
+
+	// Group is the block device group.
+        Group    []BlockDeviceGroup   `json:"group"`
+
+        // PoolSpec is specification of pool.
+	PoolSpec CStorPoolAttr `json:"poolSpec"`
+}
+
+// BlockDeviceGroup contains a collection of block devices for a given pool
+// topology in CSP.
+type BlockDeviceGroup struct {
+
+	// Item contains a list of CspBlockDevices.
+	Item []CspBlockDevice `json:"blockDevices"
+
+        // Name is the name of the block device group.
+	Name string    `json:"name"`
+}
+
+// CspBlockDevice contains the details of block device present on CSP.
+type CspBlockDevice struct {
+
+    // ID is the id where this block device is put in cspc
+    // This ID is not the property of block device and also not derived from
+    // block device.
+    // This ID can be considered as a slot ID for this block device.
+    ID string  `json:"ID"`
+
+    // Name is the name of the block device.
+    Name string `json:"name"`
+
+    // Capacity is the size of the block device. It is populated by
+    // Cstor-Operator.
+    Capacity string `json:"capacity"`
+
+    // DeviceID is the dev link or dev path of the block device. In case of
+    // sparse block devices, it contains the device path.
+    DeviceID string `json:"deviceID"`
+
+    // InUseByPool tells whether the block device is present on spc. If block
+    // device is present on SPC, it is true else false.
+    InUseByPool bool `json:"inUseByPool"`
+
+    // ReplaceBy contains details of block device that will replace the current
+    // block device. Populated by Cstor-Operator.
+    ReplaceBy CspBlockDeviceSuccessor `json:"replaceBy"`
+}
+
+// CspBlockDeviceSuccessor contains the details of block device that will
+// replace the current block device.
+type CspBlockDeviceSuccessor struct{
+
+    // Name is the name of the block device.
+    Name string `json:"name"`
+
+    // DeviceID is the dev link or dev path of the block device. In case of
```

Comment:
In the case of 'ReplaceBy' BD, BD should exist and these values 'DeviceID' and 'Capacity' can be picked from it. So, here, we won't be needing them.